desc/equilibrium/utils.py

```diff
@@ -182,3 +184,124 @@ def parse_axis(axis, NFP=1, sym=True, surface=None):
     else:
         raise TypeError("Got unknown axis type {}".format(axis))
     return axis
+
+
+def contract_equilibrium(eq, inner_rho, contract_profiles=True, copy=True):
+    """Contract an equilibrium so that an inner flux surface is the new boundary.
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium to contract.
+    inner_rho: float
+        rho value (<1) to contract the Equilibrium to.
+    contract_profiles :  bool
+        Whether or not to contract the profiles.
+        If True, the new profile's value at ``rho=1.0`` will be the same as the old
+        profile's value at ``rho=inner_rho``, i.e. in physical space, the new
+        profile is the same as the old profile. If the profile is a
+        ``PowerSeriesProfile`` or ``SplineProfile``, the same profile type will be
+        returned. If not one of these two classes, a ``SplineProfile`` will be returned
+        as other profile classes cannot be safely contracted.
+        If False, the new profile will have the same functional form
+        as the old profile, with no rescaling performed. This means the new equilibrium
+        has a physically different profile than the original equilibrium.
+    copy : bool
+        Whether or not to return a copy or to modify the original equilibrium.
+
+    Returns
+    -------
+    eq_inner: New Equilibrium object, contracted from the old one such that
+        eq.pressure(rho=inner_rho) = eq_inner.pressure(rho=1), and
+        eq_inner LCFS = eq's rho=inner_rho surface.
+        Note that this will not be in force balance, and so must be re-solved.
+
+    """
+    errorif(
+        not (inner_rho < 1 and inner_rho > 0),
+        ValueError,
+        f"Surface must be in the range 0 < inner_rho < 1, instead got {inner_rho}.",
+    )
+
+    def scale_profile(profile, rho):
+        if profile is None:
+            return profile
+        is_power_series = isinstance(profile, PowerSeriesProfile)
+        if contract_profiles and is_power_series:
+            # only PowerSeriesProfile both
+            # a) has a from_values
+            # b) can safely use that from_values to represent a
+            #    subset of itself.
```

Comment:
Technically the `SplineProfile` also meets these two criteria, and we shouldn't throw a warning if the profile is already a spline